dsgrid/config/date_time_dimension_config.py

```diff
@@ -50,8 +57,89 @@ def check_dataset_time_consistency(self, load_data_df):
                 f"load_data {time_col}s do not match expected times. mismatch={mismatch}"
             )
 
-    def convert_dataframe(self, df, project_time_dim):
-        # TODO: handle time zone
+    def get_time_dataframe(self):
+        time_col = self.get_timestamp_load_data_columns()
+        assert len(time_col) == 1, time_col
+        time_col = time_col[0]
+        schema = StructType([StructField(time_col, TimestampType(), False)])
+
+        if self.model.timezone in [TimeZone.LOCAL, TimeZone.NONE]:
+            # TODO: handle local time zone
+            schema = StructType(
+                [
+                    StructField(time_col, StringType(), False),
+                    StructField("timezone", StringType(), False),
+                ]
+            )
+            raise ValueError("TimeZone = LOCAL or NONE needs fixing")
+
+        model_time = self.list_expected_dataset_timestamps()
+        spark = SparkSession.builder.appName("dgrid").getOrCreate()
+        session_tz = spark.conf.get("spark.sql.session.timeZone")
+
+        # this is not required if we maintain UTC as session time
+        # if self.model.timezone.is_standard():
+        #     session_tz = get_equivalent_standard_system_timezone(session_tz)
+
+        # this is in local system time
+        df_time = spark.createDataFrame(model_time, schema=schema)
+
+        # TO BE DELETED
+        # model_time2 = model_time[1668:1668+24] # <----
+        # model_time2 = model_time[7385:7385+24]
+        # df_time2 = spark.createDataFrame(model_time2, schema=schema)
+        # test = df_time2.withColumn("UTC", F.to_utc_timestamp(F.col("timestamp"), session_tz)).withColumn("project", F.from_utc_timestamp(F.col("UTC"), self.model.timezone.tz_name))
+        # test.write.parquet("/Users/lliu2/Downloads/time_pyspark.parquet")
+        # test.toPandas().head(20)
+        # import pandas as pd; pd.DataFrame(test.collect()).head(20)
+        # breakpoint()
+
+        # convert to dataset timezone
+        df_time = self._convert_time_zone(
```

Comment:
I'm not sure about this function call. At this point, for tempo with dsgrid_conus_2022, `model_time` starts at `datetime.datetime(2012, 1, 1, 0, 0, tzinfo=zoneinfo.ZoneInfo(key='Etc/GMT+5'))`. `df_time` has those exact same absolute points in time. It isn't really converting to EST. Maybe this goes back to the question of absolute time vs local time.